### ISSUE TITLE
Improvement documentation for Kafka/MSK PrivateLink

### ISSUE THREAD

**Issue by sthm** [MaterializeInc/materialize#17343]
### Documentation request

https://materialize.com/docs/ops/network-security/privatelink/

Here are a couple of suggestions to improve the docs:

1. The docs jump right into creating resources without giving a lot of context. Initially I was confused where the target group should be created. My AWS account or some other AWS account? I think it would be helpful to have a high level overview of the different resources and how they interact. Something similar [to this diagram](https://d2908q01vomqb2.cloudfront.net/b6692ea5df920cad691c20319a6fffd7a4a766b8/2020/05/26/GSMSKPrivateLink_rev1.png) from [this blog](https://aws.amazon.com/blogs/big-data/how-goldman-sachs-builds-cross-account-connectivity-to-their-amazon-msk-clusters-with-aws-privatelink/), but maybe not as detailed.
2. Step 1e requires to add ip addresses to the respective target groups. We could add some prescriptive guidance on how to obtain these IPs.
3. Step 2b is really vague. It would be more helpful to state that the security group of the Kafka cluster needs to allow traffic in from the IPs of the ENIs of the NLB (and how to find these ENIs).
4. Step 2 states that one should make sure that health checks are passing, but this will only happen once the target groups are associated with an TCP listener (after Step 4).
5. It would be nice to have some trouble shooting guidance. In particular for the following error message: `ERROR:  error registering kafka topic for sink: error fetching metadata when creating new topic sink_topic for sink: Meta data fetch error: BrokerTransportFailure (Local: Broker transport failure): BrokerTransportFailure (Local: Broker transport failure)`. This happens, e.g., if the `AVAILABILITY ZONE` is not (or incorrectly) specified in the connection and the NLB has cross zone load balancing disabled (which is the default).

**Comment by chuck-alt-delete** [comment]
I'll add on to this:
- [ ] It would be helpful to have more realistic broker private hostnames, e.g. `ip-X-X-X-X.ec2.internal` and distinguish those hostnames from the target group names

and I will link these related docs issues as well:
- https://github.com/MaterializeInc/materialize/issues/17213
- https://github.com/MaterializeInc/materialize/issues/17212
